File: 21-classrooms/21-classrooms.py
def calc_rooms(input):
    ts = []
    for i in input:
        ts.append((i[0],1))
        ts.append((i[1],-1))
    ts.sort(key=lambda x:x[0])
    max_rooms = 0
    curr_rooms = 0
    for i in ts:
        curr_rooms += i[1]
        max_rooms = max(max_rooms,curr_rooms)
    return max_rooms
        
    
def main():
    testdata = [(30,75),(0,50),(60,150)]
    print(calc_rooms(testdata))
    return



if __name__ == "__main__":
    main()

File: 21-classrooms/21-classrooms.js
let classrooms = (tt) =>{
    let rooms ={}
    for (let [s,e] of tt){
        if (! (s in rooms)) rooms[s] = 0;
        rooms[s] += 1;
        if (! (e in rooms)) rooms[e] = 0;
        rooms[e] -= 1;
    }
    const sortObject = o => Object.keys(o).sort().reduce((r, k) => (r[k] = o[k], r), {})
    console.log(sortObject(rooms));
    let maxRooms =0, currRooms = 0;
    let vals = Object.values(sortObject(rooms));
    
    vals.forEach((v) =>{
        currRooms += v;
        maxRooms = Math.max(maxRooms,currRooms);
    })
    return maxRooms;
    
}


let test = [[30,75],[0,50],[60,150]]
console.log(classrooms(test));

// let t1 = {'1':0,'2':1};

// console.log(t1[2]+=1);